File: src/test/java/com/chyuck/algorithms/java/InsertionSortTests.java
package com.chyuck.algorithms.java;

import junit.framework.TestCase;
import org.junit.Assert;

public class InsertionSortTests extends TestCase {

    public void testSort() throws Exception {
        // arrange
        int[] array = new int[] { 5,3,8,4,3,11 };

        // act
        int[] result = InsertionSort.sort(array);

        // assert
        Assert.assertArrayEquals(new int[]{3, 3, 4, 5, 8, 11}, result);
    }

    public void testSort_EmptyArray() throws Exception {
        // arrange
        int[] array = new int[] {  };

        // act
        int[] result = InsertionSort.sort(array);

        // assert
        Assert.assertArrayEquals(new int[]{}, result);
    }
}

File: src/test/scala/com/chyuck/algorithms/scala/InsertionSortTests.scala
package com.chyuck.algorithms.scala

import org.junit.runner.RunWith
import org.scalatest.FunSuite
import org.scalatest.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class InsertionSortTests extends FunSuite {

  test("sort: (5,3,8,4,3,11)"){
    assert(InsertionSort.sort(Array(5,3,8,4,3,11)) === Array(3,3,4,5,8,11))
  }

  test("sort: empty array"){
    assert(InsertionSort.sort(Array()) === Array())
  }

  test("sort: (5)"){
    assert(InsertionSort.sort(Array(5)) === Array(5))
  }
}
